# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
# number of callbacks for white-sounding names
sum(data[data.race=='w'].call)

235.0

## 1. What test is appropriate for this problem? Does CLT apply?

The goal is to test whether race as indicated by name has a significant impact on the rate of callbacks for resumes. We can test the equality of two proportions of callback rates for both races. Since there are 4870 resumes, the central limit theorem applies and states that the difference between the two proportions is approximately a normal distribution. 

## 2. What are the null and alternate hypotheses?

The null hypothesis is that the callback rate for blacks $p_b$ equals the callback rate for whites $p_w$. The alterative hypothesis is that $p_b$ <> $p_w$.

In [5]:
# Compute the callback rates for both races
n_b = len(data[data.race=='b'])
n_w = len(data[data.race=='w'])
c_b = sum(data[data.race=='b'].call)
c_w = sum(data[data.race=='w'].call)
p_b = c_b / n_b
p_w = c_w / n_w
print('Callback rates for blacks & whites are: {:1.3f}, {:1.3f}'.format(p_b,p_w))

Callback rates for blacks & whites are: 0.064, 0.097


## 3. Compute margin of error, confidence interval, and p-value.

In [6]:
# Compute the standard error for the difference between callback rates
se = np.sqrt((p_b*(1-p_b)/n_b)+(p_w*(1-p_w)/n_w))
print('Standard error for the difference between callback rates: {:1.4f}'.format(se))

Standard error for the difference between callback rates: 0.0078


In [7]:
# Find 95% and 99% confidence intervals
crit = stats.norm.isf([0.025,0.005])
ci_95 = [p_b-p_w-crit[0]*se, p_b-p_w+crit[0]*se]
print("95% confidence interval: ({:2.3f}, {:2.3f})".format(ci_95[0],ci_95[1]))

95% confidence interval: (-0.047, -0.017)


In [8]:
ci_99 = [p_b-p_w-crit[1]*se, p_b-p_w+crit[1]*se]
print("99% confidence interval: ({:2.3f}, {:2.3f})".format(ci_99[0],ci_99[1]))

99% confidence interval: (-0.052, -0.012)


In [9]:
# compute the p-value of one-tail test
z_stat = (p_b - p_w) / se
p_value = stats.norm.cdf(z_stat)
print('The p-value for the test of equality of callback rates is: {:1.5f}'.format(p_value))

The p-value for the test of equality of callback rates is: 0.00002


## 4. Write a story describing the statistical significance in the context or the original problem.

Resumes with white names receive a callback rate of 9.7% as compared to 6.4% for black-sounding names. This difference of 3.3 percentage points is statistically significant since the p-value for the test of equality of callback rates is less than 0.001. Moreover, 99% confience interval indicates that the true callback rate difference could range from 1.2 percentage points to 5.2 percentage points. Therefore, based on the above data set we can conclude that U.S. labor market still faces racial discrmination in hiring.

## 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

The analysis shows that race as indicated by names has a significant effect on callbacks for interviews; but one cannot ignore other factors, such as relevance of skills to the job description, education, years of experience etc. To further understand the factors which impact or do not impact callback rate, a certain degree of feature engineering is required.